docs/content/how-mixins-work.md

```diff
@@ -0,0 +1,119 @@
+
+```
+mixins:
+- exec
+
+name: hello
+version: 0.1.0
+invocationImage: porter-hello:latest
+
+install:
+- description: "Say Hello"
+  exec:
+    command: bash
+    arguments:
+      - -c
+      - echo Hello World
+```
+
+## Mixin API
+
+Porter provides a contract that mixins must fulfill in order to be included in the Porter ecosystem. This contract specifies how Porter will execute the mixins, as show above, but also specifies how the mixin is used to build the invocation image for the bundle. Additionally, the contract specifies how a mixin can specify the inputs it can accept and the outputs that it can provide.
+
+### Build Time
+
+The previous example introduced how mixins are used when a bundle is executed. In the case of the `exec` mixin, the resulting bundle invocation image already has everything needed to execute the mixin. Other mixins may require additional runtime software. The `helm` mixin, for example, requires the [Helm](https://helm.sh/) client at runtime. Porter is responsible for building the invocation image for the bundle, so it needs to know what each mixin will need so that it can be included in the bundle. The mixin is responsible for providing any relevant lines to ensure that the generated invocation image Dockerfile has all required runtime components.
+
+In order to provide these lines, Porter expects that a mixin will provide a `build` command. This command should output any necessary Dockerfile commands to standard out. To see this in action, consider the `helm` mixin:
+
+```console
+$ ./bin/mixins/helm/helm build
+RUN apt-get update && \
+ apt-get install -y curl && \
+ curl -o helm.tgz https://storage.googleapis.com/kubernetes-helm/helm-v2.11.0-linux-amd64.tar.gz && \
+ tar -xzf helm.tgz && \
+ mv linux-amd64/helm /usr/local/bin && \
+ rm helm.tgz
+RUN helm init --client-only
+```
+
+In this case, the `helm` mixin will first run apt-get update and then install `curl` in the Docker image. Next, it will use curl to fetch Helm 2.11.0, extract it into the Docker filesystem and finally run `helm init --client-only` to do necessary setup inside the image.
+
+### Run Time
+
+The [CNAB specification](https://github.com/deislabs/cnab-spec/blob/master/103-bundle-runtime.md) specifies three actions that an invocation image should support:
+
+* install
+* upgrade
+* uninstall
+
+Porter in turn, expects that a mixin should provide a command that corresponds to each of these actions. If the corresponding action is not relevant, the mixin should still provide a command for the action and return no error. Here is `helm` mixin again for reference:
+
+```
+$ ./bin/mixins/helm/helm
+A helm mixin for porter 👩🏽‍✈️
+
+Usage:
+  helm [command]
+
+Available Commands:
+  build       Generate Dockerfile lines for the bundle invocation image
+  help        Help about any command
+  install     Execute the install functionality of this mixin
+  install     Execute the install functionality of this mixin
```

Comment:
oops, so we have a duplicate command here (maybe a bug in the helm mixin's doc?) also the example shows that we haven't finished the helm mixin since it doesn't implement all the commands... 🤔 

No need to make the docs lie (maybe just call out that that the helm mixin isn't finished?)